perses/distributed/feptasks.py

```diff
@@ -0,0 +1,136 @@
+
+
+def update_broker_location(broker_location, backend_location=None):
+    """
+    Update the location of the broker and backend from the default.
+
+    Parameters
+    ----------
+    broker_location : str
+        the url of the broker
+    backend_location: str, optional
+        the url of the backend. If none, broker_location is used.
+    """
+    if backend_location is None:
+        backend_location = broker_location
+    app.conf.update(broker=broker_location, backend=broker_location)
+
+
+class NonequilibriumSwitchTask(celery.Task):
+    """
+    This is a base class for nonequilibrium switching tasks.
+    """
+
+    def __init__(self):
+        platform = openmm.Platform.getPlatformByName("OpenCL")
+        self._cache = cache.ContextCache(platform=platform)
+
+def restore_properties(integrator, measure_shadow_work=False, measure_heat=False, metropolized_integrator=False):
+    """
+    This is a temporary utility function to restore the properties of the AlchemicalNonequilibriumLangevin...
+    integrator so that we can avoid exceptions.
+
+    Parameters
+    ----------
+    integrator : AlchemicalNone...
+        integrator to which the attributes should be added
+    measure_shadow_work : bool, default False
+        whether to tell it to measure shadow work
+    measure_heat : bool, default False
+        whether to tell it to measure heat
+    metropolized_integrator : bool, default False
+        whether the integrator is metropolized
+
+    Returns
+    -------
+    integrator : AlchemicalNone...
+        integrator with attributes added
+    """
+    integrator._measure_shadow_work = measure_shadow_work
+    integrator._measure_heat = measure_heat
+    integrator._metropolized_integrator = metropolized_integrator
+    return integrator
+
+
+@app.task(bind=True, base=NonequilibriumSwitchTask, serializer="pickle")
+def run_protocol(self, starting_positions, nsteps, thermodynamic_state, integrator):
+    """
+    Perform a switching protocol and return the nonequilibrium switching weight
+
+    Returns
+    -------
+    weight : float64
+        The nonequilibrium switching weight
+    """
+    integrator = restore_properties(integrator)
+    switching_ctx, integrator_neq = self._cache.get_context(thermodynamic_state, integrator)
+    integrator_neq = restore_properties(integrator_neq)
+    switching_ctx.setPositions(starting_positions)
+    integrator_neq.step(nsteps)
+    work = integrator_neq.get_protocol_work(dimensionless=True)
+    integrator_neq.reset()
+    return work
+
+@app.task(bind=True, base=NonequilibriumSwitchTask, serializer="pickle")
+def run_equilibrium(self, starting_positions, nsteps, lambda_state, functions, thermodynamic_state, integrator):
+    """
+    Run nsteps of equilibrium sampling at the specified thermodynamic state and return the positions.
+
+    Returns
+    -------
+    positions : [n, 3] np.ndarray quantity
+    """
+    integrator = restore_properties(integrator)
```

Comment:
No, it just adds properties back. It's a temporary hack to get around the problem of not returning the Python type.